### PR TITLE
fix(amplify-graphql-transformer-core): fixes VTL template error with \n (#9206)

### PR DESCRIPTION
**Pull request by owenr88**
#### Description of changes

Removed the line break `\n` which is breaking the template being read.

#### Description of how you validated changes

- Manual testing
- `yarn test` passes

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by lazpavel — COMMENTED** [review]
Hi @owenr88, 

Thank you for your contribution! The changes look good to me, can you add a test for the change?

File: packages/amplify-graphql-transformer-core/src/transformer-context/resolver.ts [hunk 1]
```diff
@@ -265,8 +265,9 @@ export class TransformerResolver implements TransformerResolverProvider {
     );
     if (authModes.includes(AuthorizationType.IAM)) {
       const authRoleParameter = (context.stackManager.getParameter(IAM_AUTH_ROLE_PARAMETER) as CfnParameter).valueAsString;
-      const unauthRoleParameter = (context.stackManager.getParameter(IAM_UNAUTH_ROLE_PARAMETER) as CfnParameter).valueAsString;
-      initResolver += dedent`\n
+			const unauthRoleParameter = (context.stackManager.getParameter(IAM_UNAUTH_ROLE_PARAMETER) as CfnParameter).valueAsString;
+			initResolver += '\n'
```
Comment (danielleadams) [inline packages/amplify-graphql-transformer-core/src/transformer-context/resolver.ts:269]: It looks like you're moving the line break and not removing - can you explain why this is needed? Thank you.

Comment (owenr88) [reply packages/amplify-graphql-transformer-core/src/transformer-context/resolver.ts:269]: @danielleadams Sure. `ts-dedent wasn't inserting a line break, it was inserting `\n` as plain text. There are a bunch of [issues](https://github.com/dmnd/dedent/issues?q=is%3Aissue) on dedent around not escaping characters, so perhaps there are some odd quirks around how it parses the string.

**Comment by owenr88** [comment]
> Hi @owenr88,
> 
> Thank you for your contribution! The changes look good to me, can you add a test for the change?

Yep will do! Will get it done this week

**Comment by siegerts** [comment]
Hi @owenr88, just following up on @lazpavel's note above. Any thoughts?

**Comment by danielleadams** [comment]
Hi @owenr88 - this code has been moved to https://github.com/aws-amplify/amplify-category-api. Can we close this and you can open your PR over there? Thank you!

**Pull request closed without merging**